workflows/mettannotator.nf

```diff
@@ -164,12 +175,25 @@ workflow METTANNOTATOR {
 
     LOOKUP_KINGDOM( assemblies )
 
-    PROKKA( assemblies.join( LOOKUP_KINGDOM.out.detected_kingdom ))
+    gene_caller = channel.empty()
+
+    if ( params.bakta && val(cat LOOKUP_KINGDOM.out.detected_kingdom) == "Bacteria" ) {
```

Comment:
This is new to me `val(cat LOOKUP_KINGDOM.out.detected_kingdom)`. Does it work?

The lookup is per assembly, so this won't work when mixed (arche and bacteria) exist in the sample sheet.

I think this is what you need:
```nextflow
if (params.bakta ) {
   // I'm not sure if 3 is the right index for the branch - it[] thingy
   assemblies.join( LOOKUP_KINGDOM.out.detected_kingdom ).branch {
       bacteria: it[3] == "Bacteria"
       archea: it[3] == "Archea"
   }.set { assemblies_to_annotate }

   BAKTA_BATKA( assemblies_to_annotate.bacteria, bakta_db )

   PROKKA( assemblies_to_annotate.archea )

   // I'm not sure if this is going to work downstream
   gene_caller = BAKTA_BATKA.out.mix( PROKKA.out )

   ch_versions = ch_versions.mix(BAKTA_BATKA.out.versions.first())
   ch_versions = ch_versions.mix(PROKKA.out.versions.first())

} else {

   PROKKA( assemblies.join( LOOKUP_KINGDOM.out.detected_kingdom ) )

   ch_versions = ch_versions.mix(PROKKA.out.versions.first())

   gene_caller = PROKKA
}
```

Reply:
second iteration:
```nextflow
annotations_fna = channel.empty()
annotations_gbk = channel.empty()
annotations_faa = channel.empty()
annotations_gff = channel.empty()

if (params.bakta ) {
   // I'm not sure if 3 is the right index for the branch - it[] thingy
   assemblies.join( LOOKUP_KINGDOM.out.detected_kingdom ).branch {
       bacteria: it[3] == "Bacteria"
       archea: it[3] == "Archea"
   }.set { assemblies_to_annotate }

   BAKTA_BATKA( assemblies_to_annotate.bacteria, bakta_db )

   PROKKA( assemblies_to_annotate.archea )

   ch_versions = ch_versions.mix(BAKTA_BATKA.out.versions.first())
   ch_versions = ch_versions.mix(PROKKA.out.versions.first())

   annotations_fna = annotations_fna.mix( BAKTA_BATKA.out.fna ).mix( PROKKA.out.fna )
   annotations_gbk = annotations_gbk.mix( BAKTA_BATKA.out.gbk ).mix( PROKKA.out.gbk )
   annotations_faa = annotations_faa.mix( BAKTA_BATKA.out.faa ).mix( PROKKA.out.faa )
   annotations_gff = annotations_gff.mix( BAKTA_BATKA.out.gff ).mix( PROKKA.out.gff )

} else {

   PROKKA( assemblies.join( LOOKUP_KINGDOM.out.detected_kingdom ) )

   ch_versions = ch_versions.mix(PROKKA.out.versions.first())

   annotations_fna = PROKKA.out.fna
   annotations_gbk = PROKKA.out.gbk
   annotations_faa = PROKKA.out.faa
   annotations_gff = PROKKA.out.gff
}
```

Reply:
This requires replacing the gene_caller.out downstream